### DISCUSSION TITLE
Saturation model in ESDC2A can be incorrect

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#183] [Modeling] [answered]
@cuihantao 
Hi Hantao,

I think the ``ESDC2A`` model may be incorrect.

I think the ``self.VFE.e_str`` should be
```
e_str='(INT_y*KE + Se*INT_y) - VFE'
```
The ``ESDC2A``, the code said:
```
        self.Se = Algeb(tex_name=r"V_{out}*S_e(|V_{out}|)", info='saturation output',
                        v_str='Se0',
                        e_str='SL_z0 * (INT_y - SAT_A) ** 2 * SAT_B - Se',
                        )

        self.VFE = Algeb(info='Combined saturation feedback',
                         tex_name='V_{FE}',
                         unit='p.u.',
                         v_str='vfe0',
                         e_str='(INT_y*KE + Se) - VFE'
                         )
```

However, the control diagram shows:
![image](https://user-images.githubusercontent.com/79226045/134828760-18ae977f-36ff-45e5-9ec3-98bbd4cb6565.png)

**Comment by cuihantao** [comment, marked as answer]
I think my implementation is correct. The `self.Se` variable is actually `EFD * Se(EFD)` in the block. It was noted in the `tex_name` of `self.Se`. I also put a note before `self.Se0`. 

See page 16 in Neplan's exciter manual. https://www.neplan.ch/wp-content/uploads/2015/08/Nep_EXCITERS1.pdf: `SE = B * (U - A)** 2 / U`. Our `Se` has already been multiplied by `U`, which is `vf`.

**Reply by jinningwang** [reply]
Thanks for your reply!

The control diagram from PowerWorld has two type equations (like the attached picture), which confused me sometimes. I'll follow your implementation.

``AC8B``
![image](https://user-images.githubusercontent.com/79226045/134843742-85170860-d338-48e5-a540-65bebeec4e43.png)

**Comment by cuihantao** [comment]
The drawing in AC8B is probably an error. Se alone is just a ratio of the saturated Vf (aka Efd) over the unsaturated Vf. It should be Efd*Se that gets added to Ke.

Regards,
Hantao Cui

**Reply by jinningwang** [reply]
Thanks!

Regards,
Jinning
